core/src/main/java/org/apache/accumulo/core/iterators/system/DeletingIterator.java

```diff
@@ -105,4 +110,30 @@ public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> op
       IteratorEnvironment env) {
     throw new UnsupportedOperationException();
   }
+
+  public static SortedKeyValueIterator<Key,Value> wrap(SortedKeyValueIterator<Key,Value> source,
+      boolean propogateDeletes, Behavior behavior) {
+    switch (behavior) {
+      case PROCESS:
+        return new DeletingIterator(source, propogateDeletes);
+      case FAIL:
+        return new ServerWrappingIterator(source) {
+          @Override
+          public Key getTopKey() {
+            Key top = source.getTopKey();
+            if (top.isDeleted()) {
+              throw new IllegalStateException("Saw unexpected delete " + top);
+            }
+            return top;
+          }
+        };
+      default:
+        throw new IllegalArgumentException("Unknown behavior " + behavior);
```

Comment:
This could be an `AssertionError` instead of `IllegalArgumentException`, since this class is responsible for defining what `Behavior`s are available, it's clearly programmer error on our part if we fail to handle one... rather than an API constraint imposed on the caller. It doesn't matter that much, but sometimes I think it's helpful to treat errors differently than exceptions.